keps/74-support-argo-workflow/README.md

```diff
@@ -0,0 +1,807 @@
+# KEP-74: Support Generic Workflows
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+    - [Story 3](#story-3)
+    - [Story 4](#story-4)
+- [Design Details](#design-details)
+  - [What are the Stages for Argo Workflows?](#what-are-the-stages-for-argo-workflows)
+  - [Workflow as An Unit](#workflow-as-an-unit)
+    - [How to suspend the whole workflow](#how-to-suspend-the-whole-workflow)
+    - [Drawback and Limitations](#drawback-and-limitations)
+    - [Advantages](#advantages)
+  - [Stage as An Unit](#stage-as-an-unit)
+    - [Workflow Framework in Kueue](#workflow-framework-in-kueue)
+    - [How to suspend by stages](#how-to-suspend-by-stages)
+    - [Advantages](#advantages-1)
+    - [Drawback and Limitations](#drawback-and-limitations-1)
+  - [Task as An Unit](#task-as-an-unit)
+    - [Advantages](#advantages-2)
+    - [Drawback and Limitations](#drawback-and-limitations-2)
+- [Additional Details](#additional-details)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+<!-- /toc -->
+
+## Summary
+
+This KEP outlines the proposal to integrate Generic Workflows within Kueue, discussing the advantages 
+and disadvantages of queuing workflows at varying granularity levels, alongside detailing 
+the integration methodologies. Workflow managers like Argo Workflow and Tekton can follow the convention 
+to queue the workflows.
+
+## Motivation
+
+Workflows are pivotal components in domains like Scientific Computing and Simulations, where 
+administrators often enforce resource usage quotas for different users or departments. Currently, 
+Workflows lacks native support within Kueue.
+
+### Goals
+
+- Enable support for Generic Workflows within Kueue, allowing users to simply add a label 
+`kueue.x-k8s.io/queue-name` to their workflows and submit them initially in a suspended state.
+- Should be easily extended to support various workflow managers.
+
+### Non-Goals
+
+- KEP should not only work for Argo Workflow.
+- Support resource reuse across multiple stages (potentially discussed in another KEP). 
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a machine learning engineer, I need to preprocess data before executing a training job. My 
+workflow includes two steps: data preprocessing (which doesn't require a GPU) followed by a 
+PyTorchJob. I desire that the data preprocessing stage proceeds independently of GPU quota 
+availability.
```

Comment:
This use case can be solved directly with the DAG/conditional capability of the workflow steps, e.g. PyTorchJob can start upon the completion of data processing step.